interview-questions.md

```diff
@@ -4,49 +4,46 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. Rails follows an MVC pattern. What does that mean?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: MVC stands for Model View Controller framework and it helps separate what the viewer/user sees and how that information is actually being stored internally.
 
+  Researched answer: MVC stands for Model View Controller and it is a software design pattern commonly used for developing user interfaces which divides the related program logic into three interconnected elements.  This is done to separate internal reprsenatations of information from the ways information is presented and accepted from the user.
 
 
 2. What does it mean if an application has CRUD? What does CRUD stand for?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: If an application has CRUD, it means it has achieved the goal of being an all-around user friendly app.  CRUD stands for Create, Read, Update and Delete.  If your app is lacking one of these four functions, it's probably not the best.
 
+  Researched answer: CRUD stands for Create, Read, Update, and Delete and is shorthand for the basic functions that an application must convetionally be able to fulfill in order to be considered "complete".
 
 
 3. What is a migration? Why would you use one?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: A migration is used in Rails to make changes or adjustments to a model you have already created.  For example, if you forgot a column or need to delete or rename one, you would do it through a migration.  Migrations are also excellent records of your apps history, so you can always see what has been changed or updated.
 
+  Researched answer: Migrations are a convenient way to alter your database schema over time in a consistent way.  It uses Ruby DSL so you don't have to write SQL by hand, allowing your schema and changes to be database independent. You can think of each migration as being a new 'version' of the database.  A schema starts off with nothin in it and each migration modifies it to add or remove tables, columns, or entries.  Active Record knows how to update your schema along this timeline, bringing it form whatever point it is in the history to the latest version.  It will automatically update your db/schema.rb file to match the up-to-date structure of your database.
 
 
 4. Describe a possible relationship between a model called Person and a model called Computer. Based on your description, which model would hold the foreign key?
 
-  Your answer:
+  Your answer: I am going to base this model off of someone like myself, who is one Person who has many Computers, and I am the only user of each computer.  The relationship would be Person has_many :computers, and Computer belongs_to :person.  The Computer model would hold the foreign key, since it is the belongs_to side.  In this case the foreign key would be person_id: integer.
```

Comment:
excellent work talking out your logic for this one. rather than just saying "computer." this will help you a lot in interviews!